core/drivers/firewall/stm32_risab.c

```diff
@@ -277,6 +278,51 @@ static void set_cid_priv_conf(struct stm32_risab_pdata *risab_d,
 	}
 }
 
+static TEE_Result set_rif_registers(struct stm32_risab_pdata *risab,
+				    unsigned int reg_idx)
+{
+	struct stm32_risab_rif_conf *subr_cfg = NULL;
+
+	assert(&risab->subr_cfg[reg_idx]);
+
+	subr_cfg = &risab->subr_cfg[reg_idx];
+
+	/*
+	 * This sequence will generate an IAC if the CID filtering
+	 * configuration is inconsistent with these desired rights
+	 * to apply.
+	 */
+	if (!regs_access_granted(risab, reg_idx))
+		return TEE_ERROR_ACCESS_DENIED;
+
+	set_block_dprivcfgr(risab, subr_cfg);
+	set_block_seccfgr(risab, subr_cfg);
+
+	/*
+	 * Grant page access to some CIDs, in read and/or write, and the
+	 * necessary privilege level.
+	 */
+	set_read_conf(risab, subr_cfg);
+	set_write_conf(risab, subr_cfg);
+	set_cid_priv_conf(risab, subr_cfg);
+
+	if (virt_to_phys((void *)risab_base(risab)) != RISAB1_BASE &&
+	    virt_to_phys((void *)risab_base(risab)) != RISAB2_BASE) {
+		/* Delegate RIF configuration or not */
+		if (!is_tdcid)
+			DMSG("Cannot set %s CID config for region %u",
+			     risab->risab_name, reg_idx);
```

Comment:
Should this case return an error status?
I see it was already there before this change. Maybe to consider in a specific commit or P-R.

Reply:
No we can't because we have no way of distinguishing between a CID0 filtering configuration and the absence of a CID filtering configuration

Reply:
ok